clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -704,7 +712,12 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
         public void handle(SyncGroupResponse syncResponse,
                            RequestFuture<ByteBuffer> future) {
             Errors error = syncResponse.error();
-            if (error == Errors.NONE) {
+            if (isProtocolTypeInconsistent(syncResponse.data.protocolType())
```

Comment:
I think the error code should take precedence over this validation. Perhaps we do this only when the error is NONE? Similarly for the JoinGroup response.

Reply:
You're right. Let me do this.